Tiltfile

```diff
@@ -293,13 +293,60 @@ def enable_provider(name, debug):
             links = links,
         )
 
+        provider_yaml = get_provider(yaml)
+        print("provider_yaml: " + provider_yaml)
+        k8s_yaml(blob(provider_yaml))
+
 def find_manager(yaml):
     manifests = decode_yaml_stream(yaml)
     for m in manifests:
         if m["kind"] == "Deployment":
             return m["metadata"]["name"]
     return ""
 
+# This code reimplements cmd/clusterctl/client/repository/components.InventoryObject
+# It takes all required information from the Namespace resource.
+def get_provider(yaml):
+    manifests = decode_yaml_stream(yaml)
+
+    for m in manifests:
+        if m["kind"] == "Namespace":
+            target_namespace = m["metadata"]["name"]
+            manifest_label = m["metadata"]["labels"]["cluster.x-k8s.io/provider"]
+
+    # Calculate provider type and name from manifest_label.
+    # Examples for manifest labels: "cluster-api", "infrastructure-docker",
+    # "bootstrap-kubeadm", "control-plane-kubeadm"
+    # type is based on the prefix.
+    # name is the manifest label without the provider type prefix.
+    type = "CoreProvider"
+    name = manifest_label
+    if manifest_label.startswith('infrastructure'):
+      type = "InfrastructureProvider"
+      name = manifest_label[len('infrastructure-'):]
+    if manifest_label.startswith('bootstrap'):
+      type = "BootstrapProvider"
+      name = manifest_label[len('bootstrap-'):]
+    if manifest_label.startswith('control-plane'):
+      type = "ControlPlaneProvider"
+      name = manifest_label[len('control-plane-'):]
+
+    yaml = """
+apiVersion: clusterctl.cluster.x-k8s.io/v1alpha3
+kind: Provider
+metadata:
+  labels:
+    cluster.x-k8s.io/provider: {}
+    clusterctl.cluster.x-k8s.io: ""
+    clusterctl.cluster.x-k8s.io/core: inventory
+  name: {}
+  namespace: {}
+providerName: {}
+type: {}
+version: v1.1.99
```

Comment:
Tilt deploys images with names like this `localhost:5000/gcr.io_k8s-staging-cluster-api_cluster-api-controller:tilt-12dc895fc428b26b` so I can't use that as data source.

Maybe it's good enough to just hard-code the version (we do the same in `cmd/clusterctl/hack/create-local-repository.py`)

Reply:
Ok to have this as a const for now, we can eventually make this configurable in a follow up iteration